sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -748,6 +748,7 @@ predicate
     | NOT? kind=IN '(' expression (',' expression)* ')'
     | NOT? kind=IN '(' query ')'
     | NOT? kind=RLIKE pattern=valueExpression
+    | NOT? kind=LIKE quantifier=(ANY | ALL) ('('')' | '(' expression (',' expression)* ')')
```

Comment:
What happened previously when we didn't have `'('')' | ` here? I guessed that it was also a `Parse Exception`.

Reply:
Otherwise it will throw `AnalysisException`:
```
-- !query
select company from like_any_table where company like any ()
-- !query schema
struct<>
-- !query output
org.apache.spark.sql.AnalysisException
Invalid number of arguments for function any. Expected: 1; Found: 0; line 1 pos 54
```

Reply:
Oh, it's considered as `function`. I got it.